factory/usr/lib/systemd/system/run-mnt-snapd.mount.d/fix-dependencies.conf

```diff
@@ -0,0 +1,4 @@
+# TODO, this should probably be generated by `snap-bootstrap
+# initramfs-mounts`.
+[Unit]
+DefaultDependencies=no
```

Comment:
can we just have snap-bootstrap do this first without the rest of this PR? Seems like it would be a trivial patch to snap-bootstrap here:

https://github.com/snapcore/snapd/blob/228aa864f305bf6c89d1543b96dc00befe260861/cmd/snap-bootstrap/initramfs_systemd_mount.go#L43-L46

If at all possible, making changes in snap-bootstrap is simpler and quicker and if it doesn't break things in core-initrd can be done first and make changes here simpler/easier to reason about.

Reply:
oh actually that unit file is not exactly what we want here that unit file is generated for the other targets to depend on the thing that snap-bootstrap is mounting. It sounds like this would actually be an option to `doSystemdMountImpl` which sets `--property=DefaultDependencies=no` when calling systemd-mount, right?

Reply:
`--property=` is for mount properties. This is a unit property.

Reply:
Sorry, I have tested, that works.

Reply:
After some testing, I have figured out, this is not required. Default dependencies added are only After and Before. No new Requires or Wants. So that does not break dependencies for the switch root. I will still do some cleanup in snapd, but that should not change behavior.

Reply:
I have created snapcore/snapd#11263 but it is strictly required for this PR.

Reply:
thanks for looking into that, I reviewed that PR